### ISSUE TITLE
WSL cannot connect to localhost with Network Tunnel enabled

### ISSUE THREAD

**Issue by gpechiro** [rancher-sandbox/rancher-desktop#6515]
### Actual Behavior

When WSL Network Tunnel is enabled localhost cannot be accessed from the WSL distro

### Steps to Reproduce

1. Enable Network Tunnel in the WSL preferences
2. Start a container hosting locally from within WSL
  a. `docker run --rm -p 8080:80 nginx:stable`
3. Try to access the locally running container from within WSL
  a. In a WSL shell run `curl http://localhost:8080`

### Result

The `curl` command fails
```
$ curl http://localhost:8080
curl: (7) Failed to connect to localhost port 8080 after 0 ms: Connection refused
```

### Expected Behavior

I expect to get a response from the container running locally. This is the response I get when running `curl http://localhost:8080` with the exact same repo steps but with Network Tunnel disabled.

``` 
$ curl http://localhost:8080
<!DOCTYPE html>
<html>
<head>
<title>Welcome to nginx!</title>
<style>
html { color-scheme: light dark; }
body { width: 35em; margin: 0 auto;
font-family: Tahoma, Verdana, Arial, sans-serif; }
</style>
</head>
<body>
<h1>Welcome to nginx!</h1>
<p>If you see this page, the nginx web server is successfully installed and
working. Further configuration is required.</p>

<p>For online documentation and support please refer to
<a href="http://nginx.org/">nginx.org</a>.<br/>
Commercial support is available at
<a href="http://nginx.com/">nginx.com</a>.</p>

<p><em>Thank you for using nginx.</em></p>
</body>
</html>
```

### Additional Information

At times there have been some connection issues when running docker containers or pulling images. As a company we are required to work connected to a VPN. We also host our own private image registry that is only accessible while connected to the VPN. This could be the reason for some of the inconsistent connectivity issue we experience. Enabling the Network Tunnel fixes these issues, but having it enabled is preventing us from accessing the containers that are started from WSL.

Also, I according to [Rancher Desktop release notes](https://github.com/rancher-sandbox/rancher-desktop/releases), the Network Tunnel will be the default in version 1.13 and removed in 1.14. Soon we will not have the ability to disable the Network Tunnel.

I was able to reproduce the issue on the version of Rancher I an running, 1.12.3, and on another developer' machine who was using version 1.9.1.

### Rancher Desktop Version

1.12.3

### Rancher Desktop K8s Version

N/A

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 10 Pro 22H2

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

Yes, we use a privately host VPN

**Comment by Nino-K** [comment]
@gpechiro thanks for filing this. Currently when the network tunnel is enabled, from within WSL distro you can only access your app using the WSL IP address from the host side and not the localhost. 

e.g: 
on your windows host:
```
PS C:\Users\Nino> ipconfig

Windows IP Configuration

Ethernet adapter vEthernet (WSL):

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : fe80::2262:72:fb18:bb62%31
   IPv4 Address. . . . . . . . . . . : 172.27.176.1    <===== this address
   Subnet Mask . . . . . . . . . . . : 255.255.240.0
   Default Gateway . . . . . . . . . :
```
and then from WSL distro:

```
curl 172.27.176.1:8080 
```
It should work fine. Let me know if that works for you.

**Comment by gpechiro** [comment]
@Nino-K are there any plans to allow WSL distros using Rancher to access running sites using localhost? We have some older application that rely on hitting localhost for hot reloading and other things. It would not be that easy to change this setup for each individual developer. My concern is when this becomes a permanent setting in 1.14 and beyond.

**Comment by gpechiro** [comment]
@Nino-K Thanks for the clarification. I appreciate you working on this.

**Comment by TheAnachronism** [comment]
I tried this, but I constantly get a `curl: (56) Recv failure: Connection reset by peer` for HTTP or the following for a postgres instance:
```log
psql: error: connection to server at "172.30.160.1", port 5432 failed: server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
```

Is there something else going wrong here?

**Comment by Nino-K** [comment]
@TheAnachronism I'm not too sure what is causing the error above for you. However, you can always use the bridge device IP address within the namespace from your WSL distro. The bridge device IP address is `192.168.1.2` and it's worth noting that that this IP does not change and is assigned statically. @gpechiro It might also be better for you to use this IP address in the meantime (as a workaround) too.

**Comment by joshshannon** [comment]
I thought I was going crazy with this. It's been a big problem for me with the new Network Tunnel feature and I keep checking it on each release. I thought I was doing something wrong on my local development environment. It was not an expected behavior that I could use localhost to access a container from within Windows but not from my WSL instance.

Since Windows will change that Hyper-V IP address after every reboot, it's a bit annoying to have to keep changing the IP address to access local containers. I know I could probably duct tape a solution together with a script, but using localhost would be much easier. I'm very thankful for this ticket confirming behavior I was seeing and that it seems like there's a plan to address in a future version.

@Nino-K will this ticket be addressed in 1.14 before the "old" network stack is removed?

**Comment by Nino-K** [comment]
> @Nino-K will this ticket be addressed in 1.14 before the "old" network stack is removed?

@joshshannon Yes, the goal is to have it addressed in 1.14. Meanwhile, you can use the bridge device IP address  `192.168.1.2` which is static and should not change.

**Comment by jandubois** [comment]
> before the "old" network stack is removed?

I just wanted to point out the sequence:

* new network stack must have feature parity with legacy stack
* new network stack will become the default in the following release and old stack will be deprecated
* old network stack will be removed in a later release

So there will be at least one release that has both the old and the new stack, with the new stack supporting a superset of the features of the old stack.

The old stack will only be removed if no regressions are reported against the new stack (i.e. some functionality from the old stack is still missing or broken in the new stack).

**Comment by joshshannon** [comment]
> So there will be at least one release that has both the old and the new stack, with the new stack supporting a superset of the features of the old stack.

@jandubois that makes complete sense. I thought I remember reading 1.12 release notes that your timeline was started and 1.14 would be the release where the old network stack was removed. Has that timeline shifted? I didn't see any updates to it in the 1.13 release notes yesterday.

**Comment by joshshannon** [comment]
> Meanwhile, you can use the bridge device IP address `192.168.1.2` which is static and should not change.

@Nino-K  you're definitely right this workaround does indeed work. Sorry I missed that in your earlier comment. Thanks!

**Comment by jandubois** [comment]
> I thought I remember reading 1.12 release notes that your timeline was started and 1.14 would be the release where the old network stack was removed. Has that timeline shifted?

Yes, the timeline has shifted because the new stack hasn't reached feature parity yet (exactly this issue, that we are commenting on). It is no longer considered experimental, but not yet the default.

**Comment by mathisgauthey** [comment]
Hey there. I have an issue that I think is related.

I'm using rancher desktop, activated `Networking Tunnel` and `WSL Integration`.

Using the following docker compose : 

```
version: "3.1"
services:
    db:
        container_name: odoo-db
        image: postgres:16
        environment:
            - POSTGRES_DB=postgres
            - POSTGRES_USER=odoo
            - POSTGRES_PASSWORD=odoo
            - PGDATA=/var/lib/postgresql/data/pgdata
        volumes:
            - odoo-db-data:/var/lib/postgresql/data/
        ports:
            - "5433:5432"
        tty: true

    odoo:
        container_name: odoo-web
        build: .
        depends_on:
            - db
        ports:
            - "8069:8069"
            - "8888:8888"
        volumes:
            - odoo-web-data:/var/lib/odoo
            - ./config:/etc/odoo
            - ./addons:/mnt/extra-addons
        env_file: .env
        environment:
            - HOST=db
            - PORT=5432
            - USER=odoo
            - PASSWORD=odoo
            - PYTHONUNBUFFERED=1
            - PIP_AUTO_INSTALL=1
            - DEBUGPY_ENABLE=1
            - PUID = 1000
            - PGID = 1000
        command: /usr/bin/python3 -m debugpy --listen 0.0.0.0:8888 /usr/bin/odoo --dev all --limit-time-real 999999999 -d odoo-db -r odoo -w odoo --addons-path ${CONTAINERS_ADDONS_PATH} -i ${ADDONS_LIST} -u ${ADDONS_LIST}
        tty: true

volumes:
    odoo-web-data:
    odoo-db-data:

networks:
    default:
        name: odoo-network
```

I can, from Windows using dbeaver or VS Code database client extension, connect to my postgresql database using `localhost:5433`.

But if VS Code is running using remote WSL development, I can't access it using neither `localhost:5433`, `127.0.0.1:5433` or `0.0.0.0:5433`. 

But it worked using the port found by using `ipconfig` on Windows and looking at `Ethernet adapter vEthernet (WSL (Hyper-V firewall)):`

Is this going to be fixed ?

I am pretty sure it worked on my personal machine using Docker Desktop, hence my hard troubleshooting experience here on my work laptop. Thanks in advance for any clarification as to my issue !

**Comment by mathisgauthey** [comment]
> @mathisgauthey thanks for bringing this up, we have made some changes recently that should address this specific issue you can take a look at [here](https://github.com/rancher-sandbox/rancher-desktop/issues/6636). This change will be included in our upcoming release 1.14.

Hey there thanks for the reply, pretty cool indeed ! I'll be glad to test it out and make an issue if I find anything weird 🫡

**Comment by Nino-K** [comment]
I'm going to close this issue, please give 1.14 a try, if you encounter any further issues, feel free to re-open.

**Comment by andrewduan** [comment]
Hi @Nino-K . I am aware of that this issue has been closed. but I still facing the issue that. Here is the spec.
WSL 2,
Rancher desktop 1.14.2
When I enable Networking Tunnel, and run nerdctl compose with docker-compose file, I can see bunch of containers spinned up.
one of them was postgres, port config was 5432:5432. I tried to connect to it via DBerver by host: localhost, port:5432, connection timeout, but I could connect to it via my local IPV4 address or WSL IPV4 address. Not sure what I have missed. Our projects internally connected to postgres via localhost threw error too. ( flyway threw exception said cannot connect to postgres) 


Further information on this, if I disable Networking Tunnel, after containers spinned up in Rancher desktop, our project could run successfully but I had no way to connect to postgres via DBeaver, only way I can do is nerdctl exec -it SERVICE_NAME bash. which is very inconvenient.

Thanks in advance for any suggestions.
